### PR TITLE
mpvScripts.youtube-quality: init at unstable-2020-02-11

### PR DESCRIPTION
**Pull request by lunik1**
###### Motivation for this change

Adds the youtube-quality mpv lua script.

License is set to unfree as there is no upstream license.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125284` at 2561eb84 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/0bab1af29c810388c5991d5bdedcbd77">mpvScripts.youtube-quality</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by lunik1** [comment]
After sleeping on this would it be a better approach to use multiple outputs instead of the `oscSupport` parameter?

File: pkgs/applications/video/mpv/scripts/youtube-quality.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    cp youtube-quality.lua $out/share/mpv/scripts
+  ''
+  + lib.optionalString oscSupport
+    ''
+      cp youtube-quality-osc.lua $out/share/mpv/scripts
+    ''
+  +
+  ''
+    runHook postInstall
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/video/mpv/scripts/youtube-quality.nix:32]:
```suggestion
  installPhase = ''
    runHook preInstall
    mkdir -p $out/share/mpv/scripts
    cp youtube-quality.lua $out/share/mpv/scripts
  '' + lib.optionalString oscSupport ''
    cp youtube-quality-osc.lua $out/share/mpv/scripts
  '' + ''
    runHook postInstall
  '';
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/mpv/scripts/youtube-quality.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+, oscSupport ? false
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "mpv-playlistmanager";
```
Comment (ncfavier) [inline pkgs/applications/video/mpv/scripts/youtube-quality.nix:8]: Did you forget to change this?

Comment (Artturin) [reply pkgs/applications/video/mpv/scripts/youtube-quality.nix:8]: #126143

**Pull request merged**